Default to default logging sync=info

5chdn opened the pull request:
Incldues:

- Default to default logging `sync=info`.
- Introduce `--quiet` flag to suppress that.
- Improve help messages wording a little bit.
- Updated README.md to reflect the changes.

Rationale:

- It certainly makes sense to default to `sync=info` logging at some point. (Or to write a beautiful logger.)
- Downside is that any use of `RUST_LOG` only works in combination with the `-q` switch to suppress the default logging.

5chdn commented:
Just tested, sync=info overrides RUST_LOG=trace and only works with -q.

svyatonik commented:
I've started parity with cl:
```
RUST_LOG=secretstore=trace,secretstore_net=trace ./parity --config kovan_ss1.toml
```
and with config:
```
[parity]
chain = "kovan"
base_path = "db.kovan_ss1"

[ui]
disable = true

[rpc]
disable = true

[ipc]
disable = true

[dapps]
disable = true

[network]
port = 30303

[secretstore]
disable = false
self_secret = "6c26a76e9b31048d170873a791401c7e799a11f0cefc0171cc31a49800967509"
nodes = ["cac6c205eb06c8308d65156ff6c862c62b000b8ead121a4455a8ddeff7248128d895692136f240d5d1614dc7cc4147b1bd584bd617e30560bb872064d09ea325@127.0.0.1:8085", "6b8d9b9ecde2f0d8c7b2685dd10d181000fa0b26462674dece3096488d6b8d6e3c0e4e1262e3f0eb3b997783b3d6471c281905c5fafeb7908f3aeb7326274db6@127.0.0.1:8087"]
interface = "local"
port = 8083
http_interface = "local"
http_port = 8082
path = "db.kovan_ss1/secretstore"

[ipfs]
enable = false

[snapshots]
disable_periodic = true
```
So there's no `-q` in cl. But I still can see secretstore' TRACE-level messages in output:
```
2017-08-30 09:35:46  IO Worker #3 INFO import  Syncing snapshot 12/31        #0    4/25 peers   8 KiB chain 7 KiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2017-08-30 09:35:51  IO Worker #3 INFO import  Syncing snapshot 19/31        #0    4/25 peers   8 KiB chain 7 KiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
2017-08-30 09:35:56   TRACE secretstore_net  d013…1923: executing maintain procedures
2017-08-30 09:35:56   WARN secretstore_net  d013…1923: network error Connection refused (os error 111) when establishing outbound connection with 127.0.0.1:8085
2017-08-30 09:35:56   WARN secretstore_net  d013…1923: network error Connection refused (os error 111) when establishing outbound connection with 127.0.0.1:8087
2017-08-30 09:35:56  IO Worker #1 INFO import  Syncing snapshot 26/31        #0    4/25 peers   8 KiB chain 7 KiB db 0 bytes queue 10 KiB sync  RPC:  0 conn,  0 req/s,   0 µs
```
I maybe missing something, but looks like it works even without '-q' flag

5chdn commented:
I'm talking about `pbtc -q`, sorry :)

but introducing an `-l` flag like parity has, would be nice, too.

svyatonik commented:
@5chdn Hope you don't mind that I've added this change to your PR: now RUST_LOG works even without '-q' flag. So now pbtc behaves the same way as parity does

5chdn commented:
Just noticed. Compiling.

5chdn commented:
Switching to WIP. Will test it when I'm back at some place with real internet.

5chdn commented:
> @5chdn Hope you don't mind that I've added this change to your PR: now RUST_LOG works even without '-q' flag. So now pbtc behaves the same way as parity does

Yes, works for me. I updated the README, this is good to go now. :shipit:

svyatonik commented:
Cool, thanks :)

5chdn commented on logs/src/lib.rs:
```diff
@@ -46,7 +47,12 @@ impl LogFormatter for DateAndColorLogFormatter {
 	let mut builder = LogBuilder::new();
-	builder.parse(filters);
+	let filters = match env::var("RUST_LOG") {
+		Ok(env_filters) => format!("{},{}", env_filters, filters),
```
Just to understand this, where does `env_filters` come from in this context?

svyatonik replied:
from environment variable `RUST_LOG` . and in `format!()` I just merge these two together, so that it would be either (RUST_LOG and sync=info) or (RUST_LOG), depending on `-q` existance => RUST_LOG always works

5chdn replied:
Super elegant. Thanks.
